sys/net/gnrc/netif/gnrc_netif_raw.c

```diff
@@ -104,13 +104,23 @@ static gnrc_pktsnip_t *_recv(gnrc_netif_t *netif)
     return pkt;
 }
 
+static gnrc_pktsnip_t *_remove_pkt_head(gnrc_netif_t *netif, gnrc_pktsnip_t *pkt)
+{
+    if (gnrc_netif_netdev_legacy_api(netif)) {
+        /* we don't need the netif snip: remove it */
+        return gnrc_pktbuf_remove_snip(pkt, pkt);
+    } else {
+        /* _tx_done() will free the entire list */
+        return pkt->next;
```

Comment:
Wait... what happens now to `pkt` in this case?

Reply:
Or maybe, it shouldn't be "remove head", because now, after reading a bit up on it, you are actually just saying „ok, for legacy API, I actually need to throw away the first snip, but for new API, keep it, but hand the next snip to driver->send()“. I think it it should just be `_skip_pkt_head()`, otherwise later readers might be as confused as I am.

Reply:
The pointer to the start of the pktskip chain is stored [here](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/netif/gnrc_netif.c#L1919), later retrived [here](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/netif/gnrc_netif.c#L1839) to be released in [`_tx_done()`](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/netif/gnrc_netif.c#L1823).

Reply:
> I think it it should just be `_skip_pkt_head()`, otherwise later readers might be as confused as I am.

sounds good to me!